flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/filesystem/FileSystemTableFactory.java

```diff
@@ -0,0 +1,143 @@
+import static org.apache.flink.table.descriptors.Schema.SCHEMA;
+
+/**
+ * File system {@link TableFactory}.
+ *
+ * <p>1.The partition information should be in the file system path, whether it's a temporary
+ * table or a catalog table.
+ * 2.Support insert into (append) and insert overwrite.
+ * 3.Support static and dynamic partition inserting.
+ *
+ * <p>Migrate to new source/sink interface after FLIP-95 is ready.
+ */
+public class FileSystemTableFactory implements
+		TableSourceFactory<BaseRow>,
+		TableSinkFactory<BaseRow> {
+
+	public static final String CONNECTOR_VALUE = "filesystem";
+
+	/**
+	 * Not use "connector.path" because:
+	 * 1.Using "connector.path" will conflict with current batch csv source and batch csv sink.
+	 * 2.This is compatible with FLIP-122.
+	 */
+	public static final String PATH = "path";
+
+	/**
+	 * Move these properties to validator after FLINK-16904.
+	 */
+	public static final ConfigOption<String> PARTITION_DEFAULT_NAME = key("partition.default-name")
+			.stringType()
+			.defaultValue("__DEFAULT_PARTITION__")
+			.withDescription("The default partition name in case the dynamic partition" +
+					" column value is null/empty string");
+
+	@Override
+	public Map<String, String> requiredContext() {
+		Map<String, String> context = new HashMap<>();
+		context.put(CONNECTOR, CONNECTOR_VALUE);
+		return context;
+	}
+
+	@Override
+	public List<String> supportedProperties() {
+		List<String> properties = new ArrayList<>();
+
+		// path
+		properties.add(PATH);
+
+		// schema
+		properties.add(SCHEMA + ".#." + DescriptorProperties.TABLE_SCHEMA_DATA_TYPE);
+		properties.add(SCHEMA + ".#." + DescriptorProperties.TABLE_SCHEMA_NAME);
+
+		properties.add(PARTITION_DEFAULT_NAME.key());
+
+		// format
+		properties.add(FORMAT);
+		properties.add(FORMAT + ".*");
+
+		return properties;
+	}
+
+	@Override
+	public TableSource<BaseRow> createTableSource(TableSourceFactory.Context context) {
+		DescriptorProperties properties = new DescriptorProperties();
+		properties.putProperties(context.getTable().getProperties());
+
+		return new FileSystemTableSource(
+				context.getTable().getSchema(),
+				new Path(properties.getString(PATH)),
+				context.getTable().getPartitionKeys(),
+				getPartitionDefaultName(properties),
+				getFormatProperties(context.getTable().getProperties()));
+	}
+
+	@Override
+	public TableSink<BaseRow> createTableSink(TableSinkFactory.Context context) {
+		DescriptorProperties properties = new DescriptorProperties();
+		properties.putProperties(context.getTable().getProperties());
+
+		return new FileSystemTableSink(
+				context.getTable().getSchema(),
+				new Path(properties.getString(PATH)),
+				context.getTable().getPartitionKeys(),
+				getPartitionDefaultName(properties),
+				getFormatProperties(context.getTable().getProperties()));
+	}
+
+	private static Map<String, String> getFormatProperties(Map<String, String> tableProperties) {
```

Comment:
Should we filter `FORMAT.*` properties here? otherwise we get all table connector properties.

Reply:
Don't need do this now, you can see there is a magic in `TableFactoryService.filterSupportedPropertiesFactorySpecific`.

Reply:
Oh, you're wright, I missed this logic.